framework/web/helpers/CHtml.php

```diff
@@ -1168,7 +1168,7 @@ public static function checkBoxList($name,$select,$data,$htmlOptions=array())
 
 		foreach($data as $value=>$labelTitle)
 		{
-			$checked=!is_array($select) && !strcmp($value,$select) || is_array($select) && in_array($value,$select);
+			$checked=!is_array($select) && !strcmp($value, (string)$select) || is_array($select) && in_array($value,$select);
```

Comment:
Nitpick: yii coding style has no space between arguments.

Reply:
@csears123 It doesn't make sense to use new coding standards only for bits of new code - common practise is to stick to the framework conventions unless you are doing very large changes.